lib/graphql/query/literal_input.rb

```diff
@@ -39,11 +39,11 @@ def self.defaults_for(argument_defns)
         end
       end
 
-      def self.from_arguments(ast_arguments, argument_defns, variables)
+      def self.from_arguments(ast_node, argument_defns, variables)
```

Comment:
I think this refactor was the right idea, but applied in the wrong spot. 

Instead of switching `ast_arguments` to `ast_node`, then using `ast_node.arguments_class`, we should apply the same kind of refactor the _second_ parameter in this method. 

This method should accept `argument_owner_defn` instead of `argument_defns`, then reach for `argument_owner_defn.arguments_class` at line 89 below. Then, the callers should also be updated: instead of `type.arguments,` it should pass `type`, and instead of `definition.arguments`, it should pass `definition`.

That way we can re-use the class which was derived when the schema was built. 

The thing about AST nodes is that they're constructed fresh for every parsed query. So for people who accept dynamic queries via a GraphQL API, they'll be defining a new class for each field with arguments in the query 😬 

If that refactor _works_, then I think we can remove the `arguments_class` accessor from `Language::Nodes` classes, right?